synapse/events/__init__.py

```diff
@@ -339,12 +339,19 @@ def event_id(self) -> str:
         raise NotImplementedError()
 
     @property
-    def membership(self) -> str:
-        return self.content["membership"]
+    def membership(self) -> Optional[str]:
+        return self.content.get("membership")
 
     def is_state(self) -> bool:
         return self.get_state_key() is not None
 
+    @property
+    def is_notifiable(self) -> bool:
+        return (
+            self.membership == Membership.INVITE
+            or not self.internal_metadata.is_outlier()
+        )
```

Comment:
I'm not a fan of cluttering the `Event` class with more of these "logic" style properties. 

Instead, I think we can move this to `_EventInternalMetadata` as a helper function that does

```python
    def is_notifiable(self) -> bool:
        return not self.is_outlier() or self.is_out_of_band_membership()
```

The `out_band_membership` flag is what we probably want to match on anyway.